tests/Functional/Schema/SchemaManagerFunctionalTestCase.php

```diff
@@ -579,6 +579,32 @@ public function testListForeignKeys(): void
         self::assertEquals('CASCADE', $fkeys[0]->getOption('onDelete'));
     }
 
+    public function testCreateForeignKeyGh5485(): void
+    {
+        $this->createTestTable('test_create_fk1');
+        $this->createTestTable('test_create_fk2');
+
+        $table = $this->schemaManager->listTableDetails('test_create_fk1');
+        $table->addForeignKeyConstraint(
+            'test_create_fk2',
+            ['foreign_key_test'],
+            ['id'],
+            [],
+            'i'
+        );
+        $foreignKey = $table->getForeignKeys()['i'];
+        $this->schemaManager->createForeignKey($foreignKey, $table);
```

Comment:
Passing table name as a `Table` object is deprecated (https://github.com/doctrine/dbal/pull/5529).

Reply:
only when `Table` object is passed, then functional test can be made to assert this issue

Reply:
Please do not resolve the conversation that you didn't start.

Reply:
Yes, when table is passed as string, `SqliteSchemaManager` (other DB vendors support table alter) fetches the actual indexes from the DB in https://github.com/doctrine/dbal/blob/2aec3f8da5f601462162d9b90d013554b240292b/src/Schema/SqliteSchemaManager.php#L481 and the `DROP INDEX` thus pass (yet still absolutely unneeded).

When merging up - drop the functional test completely, the SQL unit tests are enough.

Reply:
In this case, the test is invalid because the schema manager assumes that the `Table` object contains the state introspected from the database but instead it contains an arbitrary state defined in the code. This test should be definitely removed during the merge-up.